tests/state_test_util.go

```diff
@@ -48,6 +49,33 @@ import (
 // 	return nil
 // }
 
+func RunStateTestWithReader(ruleSet RuleSet, r io.Reader, skipTests []string) error {
+	bs := []byte{}
+	_, err := r.Read(bs)
+	if err != nil {
+		return err
+	}
+	var tests map[string]*StateTest
+	if err = json.Unmarshal(bs, &tests); err != nil {
+		return err
+	}
+	// results := make([]Stat)
+	for key, test := range tests {
+		for _, st := range test.Subtests() {
+			rs, ok := Rules[st.Fork]
+			if !ok {
+				return fmt.Errorf("key=%s fork=%s not supported", key, st.Fork)
+			}
+			_, err := test.Run(test, st, rs)
```

Comment:
This will also return `state`, so if desired you can add some extra debug-y `log.Println`s or even a return value to the function for handling in the cmd/ethtest package.